yarn-project/circuit-types/src/body.ts

```diff
@@ -1,13 +1,16 @@
 import { L2BlockL2Logs, TxEffect } from '@aztec/circuit-types';
 import { NUMBER_OF_L1_L2_MESSAGES_PER_ROLLUP } from '@aztec/circuits.js';
+import { makeTuple } from '@aztec/foundation/array';
+import { padArrayEnd } from '@aztec/foundation/collection';
 import { sha256 } from '@aztec/foundation/crypto';
 import { Fr } from '@aztec/foundation/fields';
-import { BufferReader, serializeToBuffer } from '@aztec/foundation/serialize';
-
-import times from 'lodash.times';
+import { BufferReader, Tuple, serializeToBuffer } from '@aztec/foundation/serialize';
 
 export class Body {
-  constructor(public l1ToL2Messages: Fr[], public txEffects: TxEffect[]) {}
+  constructor(
+    public l1ToL2Messages: Tuple<Fr, typeof NUMBER_OF_L1_L2_MESSAGES_PER_ROLLUP>,
```

Comment:
```suggestion
    /** L1 to L2 messages padded with zeros. */
    public l1ToL2Messages: Tuple<Fr, typeof NUMBER_OF_L1_L2_MESSAGES_PER_ROLLUP>,
```

Just to make it clear it's padded.